ink-wrapper-types/src/drink/client.rs

```diff
@@ -0,0 +1,160 @@
+use ::drink::{
+    frame_system, pallet_contracts,
+    runtime::{HashFor, MinimalRuntime},
+    session::Session,
+    Weight,
+};
+
+use super::*;
+use crate::util::ToAccountId;
+
+type MinimalRuntimeAccount = <MinimalRuntime as frame_system::Config>::AccountId;
+
+// NOTE: This needs to be fixed at `MinimalRuntime` as `ink-wrapper` uses `u128` to represent
+// token balances. `R::Balance` is a trait which does not provide conversion from `u128`.
+// `MinimalRuntime` has its `Balance` type fixed to `u128`.
+impl Connection<MinimalRuntime> for Session<MinimalRuntime> {
+    fn upload_code(&mut self, call: UploadCall) -> Result<HashFor<MinimalRuntime>, Error> {
+        let code_hash = self.upload(call.wasm).map_err(|_| Error::UploadFailed)?;
+        if code_hash.as_ref() != call.expected_code_hash {
```

Comment:
out of curiosity - why do we have this `expected_code_hash` at all?

Reply:
That's what aleph-client was doing, then ink-wrapper-types for aleph-client did and I just copied :) just in case this failed I guess 😅